sdk/azidentity/go.mod

```diff
@@ -2,10 +2,12 @@ module github.com/Azure/azure-sdk-for-go/sdk/azidentity
 
 go 1.18
 
+replace github.com/Azure/azure-sdk-for-go/sdk/azcore => ../azcore
```

Comment:
This change is dependent on some unreleased `azcore` functionality?

Reply:
Yes, `TokenRequestOptions.TenantID`